doc/monads/intro.md

```diff
@@ -0,0 +1,61 @@
+# Monads
+
+Monads are used heavily in Lean, as they are also in Haskell. Monads come from the wonderful world
+of [Category Theory](https://en.wikipedia.org/wiki/Monad_%28category_theory%29).
+
+Monads in Lean are so similar to Haskell that this introduction to monads is heavily based on the
+similar chapter of the [Monday Morning Haskell](https://mmhaskell.com/monads/). Many thanks to
+the authors of that material for allowing it to reused it here.
+
+Monads build on the following more fundamental type classes which you will need to understand
+first before fully understanding monads:
+
+![image](../images/monads.png)
+
+This chapter is organized to give you a bottom up introduction to monads, starting with functors and
+applicative functors to get an intuition for how these abstract structures work in Lean. Then we’ll
+tackle monads and look at some of the most common ones.
+
+## [Functor](functors.lean.md)
+A functor is a type class that provides a `map` function and the map function is something many
```

Comment:
Missing backticks in the second `map`?

Reply:
fixed in https://github.com/leanprover/lean4/pull/1505/commits/e53237c5a9afcfcfd2536283350598f393081ab3